### PR TITLE
updated JSONSchemaProps json-schema link

### LINKED ISSUE #103518: Json-schema adress link needs to be updated in generated code
#### What happened:
Update in json-schema website link address in CustomResourceDefinition is needed here: https://kubernetes.io/docs/reference/kubernetes-api/extend-resources/custom-resource-definition-v1/#JSONSchemaProps

The link is sending user to "[http://json-schema.org/)](http://json-schema.org/)"

This file is generated from the Kubernetes source code. As such, the link should be updated
in the upstream code. The source file and comment are located at:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go#L19

#### What you expected to happen:

link should direct user to "http://json-schema.org/"

#### How to reproduce it (as minimally and precisely as possible):

- generate the documentation
- read the documentation

Or look at 
https://kubernetes.io/docs/reference/kubernetes-api/extend-resources/custom-resource-definition-v1/#JSONSchemaProps

#### Environment:
- Kubernetes version (use `kubectl version`): v1.21
- Cloud provider or hardware configuration: n/a
- OS (e.g: `cat /etc/os-release`): n/a
- Kernel (e.g. `uname -a`): n/a
- Install tools: n/a
- Network plugin and version (if this is a network-related bug):
- Others:

/sig docs

### PR DESCRIPTION
**Pull request by Ritikaa96**
#### What type of PR is this?
/kind bug
/kind documentation

#### What this PR does / why we need it:

This PR corrects link address to json-schema.org in the autogenerated page we can find here: https://kubernetes.io/docs/reference/kubernetes-api/extend-resources/custom-resource-definition-v1/
Previous link was sending user to "http://json-schema.org/)"
changed it to correct address.: "http://json-schema.org/"

Fixes #103518

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```
/sig docs

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ limitations under the License.
 
 package v1
 
-// JSONSchemaProps is a JSON-Schema following Specification Draft 4 (http://json-schema.org/).
+// JSONSchemaProps is a JSON-Schema following Specification Draft 4. See http://json-schema.org/ for more details.
```
Comment (sftim) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go:19]:
```suggestion
// JSONSchemaProps is a JSON-Schema following Specification Draft 4. See https://json-schema.org/ for more details.
```
That website says:
> The current version is 2020-12! The previous version was 2019-09.

I wonder if JSONSchemaProps still follows a draft spec or if it has been updated.

Comment (Ritikaa96) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go:19]: Hi @sftim this is the specification links reference page : https://json-schema.org/specification-links.html 

2020-12 would have been Draft-8 acc to previous convention. 
Might have to investigate about what is followed . Draft 4 or latest.

Comment (Ritikaa96) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go:19]: Hi @sftim , Naming convention has been changed after Draft 7 and While investigating, I found the Changes from draft 4 to Draft 6 [here](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-01#appendix-B) it shows that 
-  Changed "id" to "$id"; all core keywords prefixed with "$"


However our file has this [here](https://github.com/kubernetes/kubernetes/blob/ec39cc2eafffa51b3267e3bd64fbd2598c0db94d/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go#L21) on L21.
 -	ID          string        `json:"id,omitempty" protobuf:"bytes,1,opt,name=id"`
 
Perhaps, the JSONSchemaProps still follows Draft spec and not the latest.

Did I get it right?

Comment (Ritikaa96) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go:19]: Also http://json-schema.org/draft-06/json-schema-release-notes.html , we can see the changes from Draft 4 to 6 and our [code](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go#L21) still has the field types changed here in Draft 6.

**Comment by palnabarun** [comment]
I checked for other instances of the link.

```
❯ rg "JSONSchemaProps is a JSON-Schema following Specification Draft"
staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types_jsonschema.go
19:// JSONSchemaProps is a JSON-Schema following Specification Draft 4 (http://json-schema.org/).

staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go
19:// JSONSchemaProps is a JSON-Schema following Specification Draft 4 (http://json-schema.org/).

staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types_jsonschema.go
19:// JSONSchemaProps is a JSON-Schema following Specification Draft 4 (http://json-schema.org/).
```

Should they also be updated?

**Comment by dims** [comment]
/retest
/assign @liggitt 

NOTE: one line fix to cleanup bad url in generated web pages. should be really quick to review. needs API reviewers approval.

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ limitations under the License.
 
 package v1
 
-// JSONSchemaProps is a JSON-Schema following Specification Draft 4 (http://json-schema.org/).
```
Comment (liggitt) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go:19]: thanks for the PR... also change it in staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types_jsonschema.go and staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types_jsonschema.go

also, the verify failure is legitimate, the openapi files need regenerating with this change

**Pull request closed without merging**